js/ui/overlay/utils.js

```diff
@@ -24,3 +26,99 @@ export const getElementMaxHeightByWindow = ($element, startLocation) => {
 
     return actualOffset * WINDOW_HEIGHT_PERCENT;
 };
+
+export const createBodyOverflowManager = () => {
+    const window = getWindow();
+    const documentElement = domAdapter.getDocument().documentElement;
+    const $body = $(domAdapter.getBody());
+
+    const isIosDevice = devices.real().platform === 'ios';
+
+    const prevSettings = {
+        overflow: null,
+        paddingRight: null,
+        position: null,
+        top: null,
+        left: null,
+    };
+
+    const setBodyPositionFixed = () => {
+        if(isDefined(prevSettings.position)) {
+            return;
+        }
+
+        const body = $body.get(0);
+        const { scrollY, scrollX } = window;
+        prevSettings.position = body.style.position;
+        prevSettings.top = body.style.top;
+        prevSettings.left = body.style.left;
+
+        $body.css({ position: 'fixed', top: `${-scrollY}px`, left: `${-scrollX}px` });
+    };
+
+    const restoreBodyPositionFixed = () => {
+        if(!isDefined(prevSettings.position)) {
+            return;
+        }
+
+        const body = $body.get(0);
+
+        const scrollY = -parseInt(body.style.top, 10);
+        const scrollX = -parseInt(body.style.left, 10);
+
+        const { position, top, left } = prevSettings;
+
+        $body.css({ position, top, left });
+
+        window.scrollTo(scrollX, scrollY);
+
+        prevSettings.position = null;
+    };
+
+    const setBodyPaddingRight = () => {
+        const scrollBarWidth = window.innerWidth - documentElement.clientWidth;
+        if(prevSettings.paddingRight || scrollBarWidth <= 0) {
+            return;
+        }
+        const paddingRight = window.getComputedStyle($body.get(0)).getPropertyValue('padding-right');
+        const computedBodyPaddingRight = parseInt(paddingRight, 10);
+        prevSettings.paddingRight = $body.css('paddingRight');
+        $body.css('paddingRight', `${computedBodyPaddingRight + scrollBarWidth}px`);
+    };
+    const restoreBodyPaddingRight = () => {
+        if(!isDefined(prevSettings.paddingRight)) {
+            return;
+        }
+        $body.css('paddingRight', prevSettings.paddingRight);
+        prevSettings.paddingRight = null;
+    };
+    const setBodyOverflow = () => {
+        if(prevSettings.overflow) {
+            return;
+        }
+        prevSettings.overflow = $body.css('overflow');
+        $body.css('overflow', 'hidden');
+    };
+    const restoreBodyOverflow = () => {
+        $body.css('overflow', prevSettings.overflow);
+        prevSettings.overflow = null;
+    };
+    return {
+        setOverflow() {
+            if(isIosDevice) {
+                setBodyPositionFixed();
+            } else {
+                setBodyPaddingRight();
+                setBodyOverflow();
+            }
+        },
+        restoreOverflow() {
+            if(isIosDevice) {
```

Comment:
On each call of the `setOverflow` and `restoreOverflow` methods checked if this device ios or not. I think it is better on cretion check this and return what need:
```
if(isIosDevice) {
 setOverflow = setBodyPositionFixed;
 restoreOverflow = restoreBodyPositionFixed;
} else {
 setOverflow = () => { 
    setBodyPaddingRight(); 
    setBodyOverflow();
}
 restoreOverflow = () => { 
    restoreBodyPaddingRight();
    restoreBodyOverflow();
 }
}

return { setOverflow, restoreOverflow }
```

Reply:
Good remark. I rewrote this part and now we doesn't check `isIosDevice` every time.